pecos/core/utils/clustering.hpp

```diff
@@ -31,6 +31,54 @@ enum {
     SKMEANS=5,
 }; /* partition_algo */
 
+enum {
+    CONSTANT_SAMPLE_SCHEDULE=0,
+    LINEAR_SAMPLE_SCHEDULE=1,
+}; /* sample strategies */
+
+extern "C" {
+    struct SampleSchedulerParam {
+        int strategy = CONSTANT_SAMPLE_SCHEDULE;
+        float sample_rate = 1.0;
```

Comment:
setting default value here may not be ideal. Consider setting it in the constructor.